internal/expr/query.go

```diff
@@ -150,9 +150,11 @@ func (qe *QueryExpr) ScanArgs(columns []string, outputArgs []any) (scanArgs []an
 	for _, column := range columns {
 		idx, ok := markerIndex(column)
 		if !ok {
-			// Columns not mentioned in output expressions are scanned into x.
-			var x any
-			ptrs = append(ptrs, &x)
+			// Columns not mentioned in output expressions are scanned into a
+			// dummy variable because sql.Scan needs pointers for all columns.
+			// We use sql.RawBytes to avoid unnecesary allocations.
+			var dummy sql.RawBytes
```

Comment:
If we don't care about the result, any reason not to use an empty struct here (`struct{}`)?

Reply:
Per the [documentation](https://pkg.go.dev/database/sql#Rows.Scan) you need to provide  `Scan` with one of the following values:
```
*string
*[]byte
*int, *int8, *int16, *int32, *int64
*uint, *uint8, *uint16, *uint32, *uint64
*bool
*float32, *float64
*interface{}
*RawBytes
*Rows (cursor value)
any type implementing Scanner (see Scanner docs)
```
and, if the provided container is compatible, it will **copy** the data to that container. The only way of avoiding copying the data is to use a pointer to `RawBytes`, if we passed an `interface{}` instead, it will always be valid but it will copy the data as well.

Reply:
If it doesn't allow the empty struct alone, we can do an empty struct that implements the Scanner interface, and does nothing with the argument.

Otherwise we're replacing one issue by the next. If you look in convert.go, you'll notice that RawBytes actually causes conversions and allocations in several cases. So it's arguably not much better than using `any` as far as allocations go.